crates/pathfinder/src/state/sync/pending.rs

```diff
@@ -1,33 +1,36 @@
 use std::sync::Arc;
 
-use pathfinder_common::BlockHash;
+use pathfinder_common::{BlockHash, BlockNumber};
 use pathfinder_storage::Storage;
 use starknet_gateway_client::GatewayApi;
-use tokio::time::Instant;
+use tokio::{sync::watch, time::Instant};
 
 use crate::state::sync::SyncEvent;
 
-/// Poll's the Sequencer's pending block and emits [pending events](SyncEvent::Pending)
-/// until the pending block is no longer connected to our current head.
-///
-/// This disconnect is detected whenever
-/// - `pending.parent_hash != head`, or
-/// - `pending` is a fully formed block and not [PendingBlock](starknet_gateway_types::reply::PendingBlock), or
-/// - the state update parent root does not match head.
-///
-/// A full block or full state update can be returned from this function if it is encountered during polling.
+/// Emits new pending data events while the current block is close to the latest block.
 pub async fn poll_pending<S: GatewayApi + Clone + Send + 'static>(
     tx_event: tokio::sync::mpsc::Sender<SyncEvent>,
     sequencer: S,
     poll_interval: std::time::Duration,
     storage: Storage,
+    latest: watch::Receiver<(BlockNumber, BlockHash)>,
+    current: watch::Receiver<(BlockNumber, BlockHash)>,
 ) {
     let mut prev_tx_count = 0;
     let mut prev_hash = BlockHash::default();
 
     loop {
         let t_fetch = Instant::now();
 
+        let latest = { *latest.borrow().0.get() };
+        let current = { *current.borrow().0.get() };
```

Comment:
The scope is because I'm worried about when the borrow's lock gets dropped.. maybe its not needed?